### DISCUSSION TITLE
Inline expansions

### DISCUSSION THREAD

**Discussion by awray3** [AllanChain/logseq-live-math#31] [Q&A] [answered]
Hello, just found this plugin and it seems great, only problem is I'm not sure how to get inline expansions working in a suitable manner.

Using the default `align` example, if I start a new block and do `cmd+shift+M` (on Mac) and start typing `align`, it inserts `$alig$` into the block before recognizing it's a macro. It then triggers and lets me write out the align elements, and the final result is 

$\begin{align*}y & =x\\ y & =x^2\end{align*}$

which fails to render because it's not inside double dollar signs.

If instead I start a new block and put in double dollar signs to start, hitting `cmd+shift+M` and starting to type `align` eats two of the dollar signs, leaving me with the same output as above.
I find that the only way to make it work is to type `$$ | $$`, where `|` is the cursor location, and then opening live math and typing align. That's kind of a lot of typing to get it to work, so I'm wondering if I'm just doing something wrong or if I can configure things differently?

Here's my settings.json:

```json
{
  "triggerSettings": null,
  "dollarTrigger": false,
  "selectEdit": false,
  "selectModifier": "<none>",
  "formatSettings": null,
  "preferDisplay": false,
  "preferMultiline": true,
  "smartFormat": true,
  "mathliveSettings": null,
  "inlinePreview": true,
  "smartFence": false,
  "smartMode": false,
  "smartSuperscript": true,
  "keybindings": [
    {
      "key": "ctrl+b",
      "command": [
        "insert",
        "\\mathbf{#@}"
      ]
    }
  ],
  "inlineShortcuts": {
    "align": "\\begin{align*}\n#? &= #? \\\\\n #? &= #?\\end{align*}"
  },
  "disableDefaultShortcuts": false,
  "disabled": false
}
```

**Comment by AllanChain** [comment, marked as answer]
By default, this plugin will use inline math mode (one dollar sign). If you want to switch to display mode (two dollar sign) temporarily, just click the title "Inline Math".
![image](https://github.com/user-attachments/assets/77a1c225-327e-4c9c-b5b7-0981aad566df)
If you want to use display mode by default, you can change it in the settings.
![image](https://github.com/user-attachments/assets/4418c1de-80f9-4700-b841-80d2e2cc4dfd)
For more details, see https://github.com/AllanChain/logseq-live-math?tab=readme-ov-file#how-to-switch-between-display-math--and-inline-math-
